dlib/dnn/loss_abstract.h

```diff
@@ -2144,6 +2144,20 @@ namespace dlib
                   in WHAT THIS OBJECT REPRESENTS for details.
         !*/
 
+        tensor& get_eccm();
+        /*!
+            ensures
+                - returns the empirical cross-correlation matrix computed by the loss.
+                - this is only meant to be used for visualization/debugging purposes.
+        !*/
```

Comment:
I am not sure if this will ever be used. But even if the user modifies the `eccm`, it doesn't really affect the loss, since it's re-computed at every step.

So, in case someone has to modify the `eccm` between training steps, they don't need to copy it.